### PR TITLE
Tighten RBAC permissions

### PR DESCRIPTION
**Pull request by kishorj**

### CODE REVIEW

File: controllers/elbv2/targetgroupbinding_controller.go [hunk 1]
```diff
@@ -67,14 +67,14 @@ type targetGroupBindingReconciler struct {
 }
 
 // +kubebuilder:rbac:groups=elbv2.k8s.aws,resources=targetgroupbindings,verbs=get;list;watch;update;patch;create;delete
-// +kubebuilder:rbac:groups=elbv2.k8s.aws,resources=targetgroupbindings/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=elbv2.k8s.aws,resources=targetgroupbindings/status,verbs=update;patch
 // +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;update;patch
```
Comment (M00nF1sh) [inline controllers/elbv2/targetgroupbinding_controller.go:71]: for pods, we should only need get/list/watch. but not update/patch. 
can we test whether without pod's update/patch, we can still patch pod status?

Comment (kishorj) [reply controllers/elbv2/targetgroupbinding_controller.go:71]: sure

Comment (kishorj) [reply controllers/elbv2/targetgroupbinding_controller.go:71]: don't need update/patch for pod for the status updates.

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Pull request merged**